### PR TITLE
[Docs][KubeRay] add a guide for deploying vLLM with RayService

### PR DESCRIPTION
**Pull request by andrewsykim**
## Why are these changes needed?

Based on sample in https://github.com/ray-project/kuberay/pull/2289 

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [X] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [X] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by andrewsykim** [comment]
@angelinalg @kevin85421 this is ready for review now

File: doc/source/cluster/kubernetes/examples/vllm-rayservice.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    --location=us-east4-c \
+    --num-nodes=2 \
+    --accelerator=type=nvidia-l4,count=2,gpu-driver-version=latest
+```
+
+This example uses L4 GPUs. Each model replica will use 2 L4 GPUs using vLLM's tensor parallelism.
+
+## Install the KubeRay Operator
+
+Deploy the KubeRay operator using Helm:
```
Comment (kevin85421) [inline doc/source/cluster/kubernetes/examples/vllm-rayservice.md:30]: Typically, the KubeRay documentation adds a hyperlink to https://docs.ray.io/en/latest/cluster/kubernetes/getting-started/raycluster-quick-start.html#step-2-deploy-a-kuberay-operator when referring to installing a KubeRay operator. Take [RayJob quick start](https://docs.ray.io/en/latest/cluster/kubernetes/getting-started/rayjob-quick-start.html#step-2-install-the-kuberay-operator) as an example,


Screenshot 2024-09-03 at 6 49 29 PM

**Review by kevin85421 — APPROVED** [review]

File: doc/source/cluster/kubernetes/examples/vllm-rayservice.md (hunk 1 above)
Comment (andrewsykim) [reply doc/source/cluster/kubernetes/examples/vllm-rayservice.md:30]: Added the link at the bottom of this section. I figured it doesn't hurt to include the steps here so users don't have to open another doc for it

Comment (kevin85421) [reply doc/source/cluster/kubernetes/examples/vllm-rayservice.md:30]: The reason I always point to a single location for deploying the KubeRay operator is to make it the sole location for the KubeRay deployment section. This way, when we need to update the documentation, we only need to update one place, avoiding multiple locations that describe how to install the KubeRay operator, which could result in different commands and versions being used.

Comment (andrewsykim) [reply doc/source/cluster/kubernetes/examples/vllm-rayservice.md:30]: That makes sense, I'll remove this part and just reference the quick start

Comment (andrewsykim) [reply doc/source/cluster/kubernetes/examples/vllm-rayservice.md]: updated

**Comment by andrewsykim** [comment]
@angelinalg can you review please?

File: doc/source/cluster/kubernetes/examples/vllm-rayservice.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+Create a GKE cluster with a GPU node pool:
+```sh
+gcloud container clusters create kuberay-gpu-cluster \
+    --machine-type=g2-standard-24 \
+    --location=us-east4-c \
+    --num-nodes=2 \
+    --accelerator=type=nvidia-l4,count=2,gpu-driver-version=latest
+```
+
+This example uses L4 GPUs. Each model replica will use 2 L4 GPUs using vLLM's tensor parallelism.
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/examples/vllm-rayservice.md:26]:
```suggestion
This example uses L4 GPUs. Each model replica uses 2 L4 GPUs using vLLM's tensor parallelism.
```

File: doc/source/cluster/kubernetes/examples/vllm-rayservice.md [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+## Create a Kubernetes Secret containing your Hugging Face access token
+
+Create a Kubernetes Secret containing your Hugging Face access token:
+```sh
+export HF_TOKEN=<Hugging Face access token>
+kubectl create secret generic hf-secret   --from-literal=hf_api_token=${HF_TOKEN}   --dry-run=client -o yaml | kubectl apply -f -
+```
+
+This secret will be referenced as an environment variable in the RayCluster used in the next steps.
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/examples/vllm-rayservice.md:41]:
```suggestion
This guide references this secret as an environment variable in the RayCluster in the next steps.
```

File: doc/source/cluster/kubernetes/examples/vllm-rayservice.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+## Deploy a RayService
+
+Create a RayService custom resource:
+```
+kubectl apply -f https://raw.githubusercontent.com/ray-project/kuberay/master/ray-operator/config/samples/vllm/ray-service.vllm.yaml
+```
+
+The RayService is configured to deploy a Ray Serve application, running vLLM as the serving engine for the Llama 3 8B Instruct model. The code used in this example can be found [here](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/vllm/serve.py).
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/examples/vllm-rayservice.md:50]:
```suggestion
This step configures RayService to deploy a Ray Serve app, running vLLM as the serving engine for the Llama 3 8B Instruct model. You can find the code for this example [on GitHub](https://github.com/ray-project/kuberay/blob/master/ray-operator/config/samples/vllm/serve.py).
```

File: doc/source/cluster/kubernetes/examples/vllm-rayservice.md [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+          # NOTE: num_gpus is set automatically based on TENSOR_PARALLELISM
+      runtime_env:
+        working_dir: "https://github.com/ray-project/kuberay/archive/master.zip"
+        pip: ["vllm==0.5.4"]
+        env_vars:
+          MODEL_ID: "meta-llama/Meta-Llama-3-8B-Instruct"
+          TENSOR_PARALLELISM: "2"
+```
+
+Wait for the RayService resource to be ready. You can inspect it's status by running the following command:
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/examples/vllm-rayservice.md:72]:
```suggestion
Wait for the RayService resource to be ready. You can inspect its status by running the following command:
```

File: doc/source/cluster/kubernetes/examples/vllm-rayservice.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        serveDeploymentStatuses:
+          VLLMDeployment:
+            healthLastUpdateTime: "2024-08-08T22:56:50Z"
+            status: HEALTHY
+        status: RUNNING
+```
+
+## Send a prompt
+
+Once you've confirmed the Ray Serve deployment is healthy, you can establish a port-forwarding session for the Serve application:
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/examples/vllm-rayservice.md:93]:
```suggestion
Confirm the Ray Serve deployment is healthy, then you can establish a port-forwarding session for the Serve app:
```

File: doc/source/cluster/kubernetes/examples/vllm-rayservice.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+```sh
+$ kubectl port-forward svc/llama-3-8b-serve-svc 8000
+```
+
+Note that this Kubernetes Service will be created after the Serve applications are ready and running.
```
Comment (angelinalg) [inline doc/source/cluster/kubernetes/examples/vllm-rayservice.md:99]:
```suggestion
Note that KubeRay creates this Kubernetes Service after the Serve apps are ready and running.
```

**Review by angelinalg — APPROVED** [review]
Just some style nits. Please consider using [Vale](https://docs.ray.io/en/latest/ray-contribute/docs.html#how-to-use-vale) to find these issues in the future. Please excuse any inaccuracies I introduced in my suggestions and correct as needed. Happy to answer any questions you have about the suggestions. Thanks for your contribution!

File: doc/source/cluster/kubernetes/examples/vllm-rayservice.md (hunk 6 above)
Comment (andrewsykim) [reply doc/source/cluster/kubernetes/examples/vllm-rayservice.md:93]: done

File: doc/source/cluster/kubernetes/examples/vllm-rayservice.md (hunk 5 above)
Comment (andrewsykim) [reply doc/source/cluster/kubernetes/examples/vllm-rayservice.md:72]: done

File: doc/source/cluster/kubernetes/examples/vllm-rayservice.md (hunk 4 above)
Comment (andrewsykim) [reply doc/source/cluster/kubernetes/examples/vllm-rayservice.md:50]: done

File: doc/source/cluster/kubernetes/examples/vllm-rayservice.md (hunk 2 above)
Comment (andrewsykim) [reply doc/source/cluster/kubernetes/examples/vllm-rayservice.md:26]: done

File: doc/source/cluster/kubernetes/examples/vllm-rayservice.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+
+This guide demonstrates how to [Serve a Large Language Model with vLLM](https://docs.ray.io/en/latest/serve/tutorials/vllm-example.html) on Kubernetes using KubeRay. The example in this guide deploys the `meta-llama/Meta-Llama-3-8B-Instruct` model from Hugging Face on Google Kubernetes Engine (GKE).
+
+## Prerequisites
+
+This example downloads model weights from Hugging Face. You will need to complete the following
+prerequisites to successfully complete this guide:
+* A [Hugging Face account](https://huggingface.co/)
+* A Hugging Face [access token](https://huggingface.co/docs/hub/security-tokens) with read access to gated repos.
+* Access to the Llama 3 8B model. This usually requires signing an agreement on Hugging Face to access this model. Go to the [Llama 3 model page](https://huggingface.co/meta-llama/Meta-Llama-3-8B) for more details.
```
[the remark this answers is not in the document]
Comment (andrewsykim) [reply doc/source/cluster/kubernetes/examples/vllm-rayservice.md:13]: done

File: doc/source/cluster/kubernetes/examples/vllm-rayservice.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
+(kuberay-vllm-rayservice-example)=
+
+# Serve a Large Language Model with vLLM on Kubernetes
+
+This guide demonstrates how to [Serve a Large Language Model with vLLM](https://docs.ray.io/en/latest/serve/tutorials/vllm-example.html) on Kubernetes using KubeRay. The example in this guide deploys the `meta-llama/Meta-Llama-3-8B-Instruct` model from Hugging Face on Google Kubernetes Engine (GKE).
+
+## Prerequisites
+
+This example downloads model weights from Hugging Face. You will need to complete the following
```
[the remark this answers is not in the document]
Comment (andrewsykim) [reply doc/source/cluster/kubernetes/examples/vllm-rayservice.md:9]: done

**Comment by andrewsykim** [comment]
Thanks for the review @angelinalg, addressed your feedback!

**Pull request merged**